### ISSUE TITLE
Bug: [key-spacing] and [type-annotation-spacing] are in conflict

### ISSUE THREAD

**Issue by sxyazi** [typescript-eslint/typescript-eslint#6981]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=JYOwLgpgTgZghgYwgAgGIHt3IN4ChnJwBcyIArgLYBG0+hcxplNUuAvrkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Aa0UNuWIEMyzAObowAbUiJo0APbRIAGkyQ+jIU1GQyM+DI0BfALoLw2PEVLIK1Og2ZsLiWnyZMZ+Pvkp6e-QUxEMSWk5SFN9EH0gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEAviDUA

### Repro Code

```typescript
interface Foo {
  a: number
  aaa: number
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/key-spacing": ["error", {"align": "colon"}],
    "@typescript-eslint/type-annotation-spacing": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
  }
}
```

### Expected Result

`type-annotation-spacing` can be configured with multiple space support or as an escape that allows disabling rules partly in favor of `key-spacing`, e.g.

```json
{
  "@typescript-eslint/type-annotation-spacing": [
    "error",
    "overrides": { "property": false }
  ]
}
```

### Actual Result

When I enforce some spaces after the colon of a property to align it with others, then a conflict with the `type-annotation-spacing` rule raised.

<img width="637" alt="issue-1" src="https://user-images.githubusercontent.com/17523360/235634489-65412b3a-7b21-451e-bdbf-569943229a16.png">

<img width="646" alt="issus-2" src="https://user-images.githubusercontent.com/17523360/235634512-600ac3fe-496b-4a74-9111-a9ba1483208b.png">

### Additional Info

I found [an issue](https://github.com/typescript-eslint/typescript-eslint/pull/4600) that related to.

**Comment by bradzacher** [comment]
It's entirely acceptable and expected that you can configure rules so that they conflict.
Its up to you to adjust your config so that you do not have conflicts.

In this example you definitely do not want the `type-annotation-spacing` rule because it designed to always enforce there is or isn't a space, but the `key-spacing` rule enforces that there sometimes is or sometimes isn't. I.e. The rules will always conflict.

**Comment by sxyazi** [comment]
Thanks for the quick reply.

As far as I know that `type-annotation-spacing` has a larger impact than `key-spacing` (not just for object keys only), and disabling it will result in the following code not applying the rule:

```ts
let foo: string = "bar".
```

So it might be better to partially disable `type-annotation-spacing` or make it compatible with `key-spacing`.
I'm sorry if I got something wrong.
